### PR TITLE
[MCH-268-132] refactor: threshold on config

### PR DESCRIPTION
**Pull request by gotzenx**

### CODE REVIEW

File: packages/yswaps/scripts/configs/mainnet.ts [hunk 1]
```diff
@@ -47,10 +51,12 @@ const mainnetConfig: StrategyConfiguration<'MAINNET'> = {
           {
             tokenIn: '0xD533a949740bb3306d119CC777fa900bA034cd52', // CRV
             tokenOut: '0x29059568bB40344487d62f7450E78b8E6C74e0e5', // crvYfiEth
+            threshold: utils.parseEther('1'),
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/configs/mainnet.ts:54]: If you could set this to a "40k threshold" in terms of CRV would be 10/10. Same with tokens above.

File: packages/yswaps/scripts/libraries/utils/should-execute-trade.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const token = await ethers.getContractAt<IERC20>(IERC20__factory.abi, trade.tokenIn, randomAddress);
+    const balance = await token.balanceOf(strategy);
+    if (balance.gt(trade.threshold)) executeConfirmations.push(true);
+  }
+
+  // Execute only of every  trade should execute.
+  if (checkType === 'total') return executeConfirmations.length === trades.length;
+
+  // Execute if at least one trade should execute.
+  return !!executeConfirmations.length;
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/libraries/utils/should-execute-trade.ts:27]:
```suggestion
  return executeConfirmations.length > 0;
```
A little bit more explicit. `!!` is more of a boolean operator.

File: packages/yswaps/scripts/libraries/utils/should-execute-trade.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  checkType,
+}: {
+  strategy: string;
+  trades: SimpleEnabledTrade[];
+  checkType: 'total' | 'partial';
+}): Promise<boolean> {
+  const executeConfirmations: boolean[] = [];
+  const randomAddress = await wallet.generateRandomAddress();
+  for (const trade of trades) {
+    const token = await ethers.getContractAt<IERC20>(IERC20__factory.abi, trade.tokenIn, randomAddress);
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/libraries/utils/should-execute-trade.ts:18]:
```suggestion
  for (const trade of trades) {
    const token = await ethers.getContractAt<IERC20>(IERC20__factory.abi, trade.tokenIn);
```
There should be no need of setting the signer if you are using `getContractAt` is just needed when you do `IERC20__factory.connect(address, signer)`

**Review by b0dhidharma — APPROVED** [review]
Great work dude ! 🚀

File: packages/yswaps/environment.d.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { SUPPORTED_NETWORKS } from '../commons/utils/network';
+declare namespace NodeJS {
+  export interface ProcessEnv {
+    HARDHAT_DEPLOY_FORK: SUPPORTED_NETWORKS;
+  }
+}
```
Comment (b0dhidharma) [inline packages/yswaps/environment.d.ts:6]: 🤯

File: packages/yswaps/scripts/configs/mainnet.ts [hunk 5]
```diff
@@ -8,41 +8,41 @@ import { utils } from 'ethers';
...
+          {
+            tokenIn: '0x4e3FBD56CD56c3e72c1403e103b45Db9da5B9D2B', // CVX
+            tokenOut: '0x8282BD15dcA2EA2bDf24163E8f2781B30C43A2ef', // crvSpellEth
+            threshold: utils.parseEther('1250'),
+          },
+        ],
+        solver: 'CurveSpellEth',
+      },
+    ],
+  },
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/configs/mainnet.ts:45]: Are these enabled in mainnet?

**Review by b0dhidharma — APPROVED** [review]
Noiceee !

File: packages/yswaps/scripts/configs/mainnet.ts (hunk 5 above)
Comment (gotzenx) [reply packages/yswaps/scripts/configs/mainnet.ts]: back to commented

**Pull request merged**